pkg/pillar/agentlog/logobject.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package agentlog
+
+import (
+	"encoding/json"
+	log "github.com/sirupsen/logrus"
+)
+
+// LogObject : Holds all key values pairs to be logged later
+type LogObject struct {
+	fields map[string]interface{}
+}
+
+// NewLogObject :
+// objType -> log, action, relation etc
```

Comment:
Can we make this into an enum so that folks don't use unconstrained strings for objType?

Reply:
Should we have this enumeration inside in the current file (logobject.go)?

Reply:
I think you can. Just need to pick the initial set of things you want to describe.
It could be that some will be associated with specific functions e.g., a relation might have a function with (fromName, fromUUID, toName, toUUID) as arguments hence more specific than the generic LogObject. But the enum values still make sense.

Reply:
will do